sklearn/preprocessing/data.py

```diff
@@ -2472,3 +2475,239 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
+    a sparse matrix where each column corresponds to one possible value of one
+    feature. It is assumed that input features take on values in the range
+    [0, n_values).
+
+    This encoding is needed for feeding ordinal features to many scikit-learn
+    estimators, notably linear models and SVMs with the standard kernels.
+
+    Read more in the :ref:`User Guide <preprocessing_ordinal_features>`.
+
+    Parameters
+    ----------
+    n_values : 'auto', int or array of ints
+        Number of values per feature.
+
+        - 'auto' : determine value range from training data.
+        - int : number of ordinal values per feature.
+                Each feature value should be in ``range(n_values)``
+        - array : ``n_values[i]`` is the number of ordinal values in
+                  ``X[:, i]``. Each feature value should be
+                  in ``range(n_values[i])``
+
+    ordinal_features : "all" or array of indices or mask
+        Specify what features are treated as ordinal.
+
+        - 'all' (default): All features are treated as ordinal.
+        - array of indices: Array of ordinal feature indices.
+        - mask: Array of length n_features and with dtype=bool.
+
+        Non-ordinal features are always stacked to the right of the matrix.
+
+    dtype : number type, default=np.float
+        Desired dtype of output.
+
+    sparse : boolean, default=True
+        Will return sparse matrix if set True else will return an array.
+
+    handle_unknown : str, 'error' or 'ignore'
+        Whether to raise an error or ignore if a unknown ordinal feature is
+        present during transform.
+
+    Attributes
+    ----------
+    active_features_ : array
+        Indices for active features, meaning values that actually occur
+        in the training set. All featurs are available when n_values is
+        ``'auto'``.
+
+    feature_indices_ : array of shape (n_features,)
+        Indices to feature ranges.
+        Feature ``i`` in the original data is mapped to features
+        from ``feature_indices_[i]`` to ``feature_indices_[i+1]``
+        (and then potentially masked by `active_features_` afterwards)
+
+    n_values_ : array of shape (n_features,)
+        Maximum number of values per feature.
+
+    Examples
+    --------
+    Given a dataset with three features and four samples, we let the encoder
+    find the maximum value per feature and transform the data to a binary
+    unary encoding.
+
+    >>> from sklearn.preprocessing import UnaryEncoder
+    >>> enc = UnaryEncoder()
+    >>> enc.fit([[0, 0, 3], [1, 1, 0], [0, 2, 1], \
+[1, 0, 2]])  # doctest: +ELLIPSIS
+    UnaryEncoder(dtype=<... 'numpy.float64'>, handle_unknown='error',
+           n_values='auto', ordinal_features='all', sparse=True)
+    >>> enc.n_values_
+    array([2, 3, 4])
+    >>> enc.feature_indices_
+    array([0, 1, 3, 6])
+    >>> enc.active_features_
+    array([0, 1, 2, 3, 4, 5])
+    >>> enc.transform([[0, 1, 1]]).toarray()
```

Comment:
It's not very useful to transform only 1s. Use a 2 in there, please.

Reply:
done